server/server.go

```diff
@@ -1440,9 +1440,6 @@ func (server *ArgoCDServer) newStaticAssetsHandler() func(http.ResponseWriter, *
 		} else {
 			if isMainJsBundle(r.URL) {
 				cacheControl := "public, max-age=31536000, immutable"
-				if !fileRequest {
```

Comment:
After Go1.23, when the response status isn't 200, the some header is deleted.
ref. https://tip.golang.org/doc/go1.23

Reply:
@agaudreault is this change unnecessary now that we've merged this? https://github.com/argoproj/argo-cd/pull/21748/files#diff-8b7750f45b61f757e4f73dd156bf442799203de80ebd846badcc3c8addfc4017L1450

Reply:
That is correct

Reply:
@sivchari can you revert this change? After that, the upgrade lgtm!

Reply:
@crenshaw-dev 
Make sense to me. Fixed.